bambi/models.py

```diff
@@ -225,21 +227,57 @@ def __init__(
         # Build priors
         self._build_priors()
 
+    @overload
     def fit(
         self,
-        draws=1000,
-        tune=1000,
-        discard_tuned_samples=True,
-        omit_offsets=True,
-        include_mean=False,
-        inference_method="mcmc",
-        init="auto",
-        n_init=50000,
-        chains=None,
-        cores=None,
-        random_seed=None,
+        draws: int = 1000,
+        tune: int = 1000,
+        discard_tuned_samples: bool = True,
+        omit_offsets: bool = True,
+        include_mean: bool = False,
+        inference_method: Literal["mcmc", "blackjax_nuts", "numpyro_nuts", "laplace"] = "mcmc",
```

Comment:
One problem I see here is that after #775, users have access to a huge variety of samplers through bayeux. I'm not sure how this is handled with literals. But, is it possible that what we need is to create a Literal on the fly (at import time) where we check whether users have access to bayeux, and depending on that, the Literal contains different types?

Also, I don't think there's a convenient way to access the list of samplers without first creating a model. I'm going to create an issue now.